google/cloud/aiplatform/_pipeline_based_service/pipeline_based_service.py

```diff
@@ -0,0 +1,328 @@
+                Specifies the service account for workload run-as account.
+                Users submitting jobs must have act-as permission on this run-as account.
+            network (str):
+                The full name of the Compute Engine network to which the job
+                should be peered. For example, projects/12345/global/networks/myVPC.
+                Private services access must already be configured for the network.
+                If left unspecified, the job is not peered with any network.
+            project (str):
+                Optional. The project to run this PipelineJob in. If not set,
+                the project set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to create PipelineJob. If not set,
+                location set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to create the PipelineJob.
+                Overrides credentials set in aiplatform.init.
+            experiment (Union[str, experiments_resource.Experiment]):
+                Optional. The Vertex AI experiment name or instance to associate to the PipelineJob executing
+                this model evaluation job.
+        Returns:
+            (VertexAiPipelineBasedService):
+                Instantiated representation of a Vertex AI Pipeline based service.
+        """
+
+        if not display_name:
+            display_name = cls._generate_display_name()
+
+        self = cls._empty_constructor(
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        service_pipeline_job = pipeline_jobs.PipelineJob(
+            display_name=display_name,
+            template_path=template_path,
+            job_id=job_id,
+            pipeline_root=pipeline_root,
+            parameter_values=template_params,
+            encryption_spec_key_name=encryption_spec_key_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        # Suppresses logs from PipelineJob
+        # The class implementing _VertexAiPipelineBasedService should define a custom log message
+        logging.getLogger("google.cloud.aiplatform.pipeline_jobs").setLevel(
+            logging.WARNING
+        )
+
+        service_pipeline_job.submit(
+            service_account=service_account,
+            network=network,
+            experiment=experiment,
+        )
+
+        logging.getLogger("google.cloud.aiplatform.pipeline_jobs").setLevel(
+            logging.INFO
+        )
+
+        self._gca_resource = service_pipeline_job.gca_resource
+
+        return self
+
+    @classmethod
+    def list(
+        cls,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[str] = None,
+    ) -> List["pipeline_jobs.PipelineJob"]:
```

Comment:
This seems like it should return a list of `_VertexAiPipelineBasedService` instead of `PipelineJobs`

Reply:
It calls the `PipelineJob` list method (see line 312). I could either remove `@classmethod` from this and leave the current implementation or change line 323 to instantiate the returned `PipelineJob` as a `_VertexAiPipelineBasedService` before appending to the returned list.

Reply:
Preference for the second option.

Reply:
Updated to return a list of `_VertexAiPipelineBasedService` instances. Also added a unit test for this method.